volatility3/framework/symbols/linux/utilities/module_extract.py

```diff
@@ -0,0 +1,891 @@
+        e_flags = b"\x00\x00\x00\x00"
+        e_ehsize = b"\x34\x00"  # 52 - size of ELF Header
+        e_phentsize = b"\x00\x00"
+        e_phnum = b"\x00\x00"
+        e_shentsize = b"\x28\x00"
+        e_shnum = struct.pack("<H", num_sections + 1)
+        e_shstrndx = struct.pack("<H", num_sections)
+
+        header = (
+            e_ident
+            + e_type
+            + e_machine
+            + e_version
+            + e_entry
+            + e_phoff
+            + e_shoff
+            + e_flags
+            + e_ehsize
+            + e_phentsize
+            + e_phnum
+            + e_shentsize
+            + e_shnum
+            + e_shstrndx
+        )
+
+        # should never happen as we make the header ourselves
+        if len(header) != 52:
+            vollog.error(
+                f"_get_header_32 created a header of {len(header)} bytes. Cannot proceed"
+            )
+            return None
+
+        return header
+
+    @classmethod
+    def _make_elf_header_64(
+        cls, sect_hdr_offset: int, num_sections: int
+    ) -> Optional[bytes]:
+        """
+        Creates a 64 bit ELF header for the file based on recovered values
+        Called last as it needs information computed from the sections
+
+        Spec: https://refspecs.linuxfoundation.org/elf/gabi4+/ch4.eheader.html
+        """
+        e_ident = b"\x7f\x45\x4c\x46\x02\x01\x01\x00\x00\x00\x00\x00\x00\x00\x00\x00"
+        e_type = b"\x01\x00"  # relocateble
+        e_machine = b"\x3e\x00"  # EM_X86_64
+        e_version = b"\x01\x00\x00\x00"
+        e_entry = b"\x00" * 8  # The .init sections are freed after module load
+        e_phoff = b"\x00" * 8
+        e_shoff = struct.pack("<Q", sect_hdr_offset)
+        e_flags = b"\x00\x00\x00\x00"
+        e_ehsize = b"\x40\x00"  # 64 / size of ELF Header
+        e_phentsize = b"\x00\x00"
+        e_phnum = b"\x00\x00"
+        e_shentsize = b"\x40\x00"  # 64
+        e_shnum = struct.pack("<H", num_sections + 1)
+        e_shstrndx = struct.pack("<H", num_sections)
+
+        header = (
+            e_ident
+            + e_type
+            + e_machine
+            + e_version
+            + e_entry
+            + e_phoff
+            + e_shoff
+            + e_flags
+            + e_ehsize
+            + e_phentsize
+            + e_phnum
+            + e_shentsize
+            + e_shnum
+            + e_shstrndx
+        )
+
+        # should never happen as we make the header ourselves
+        if len(header) != 64:
+            vollog.error(
+                f"_get_header_64 created a header of {len(header)} bytes. Cannot proceed"
+            )
+            return None
+
+        return header
+
+    @classmethod
+    def _calc_sect_type(cls, section_name: str) -> Optional[int]:
+        """
+        This function makes a best effort to map common section names
+        to their attributes
+        """
+        known_sections = {
+            ".note.gnu.build-id": linux_constants.SHT_NOTE,
```

Comment:
Yeah, so... we do use constants already...  5;P

Reply:
Good catch. Will move although I am not sure where else its useful except to map sections to their section type, but its not a good pattern to encourage.

Reply:
No, no, this is fine, it was more in relation to the fact that we using `constants` here, so we should use them for the ELF data too!  This is fine, it was just an example of us already using constants!